playbooks/roles/dev-patcher/tasks/patch-repos.yml

```diff
@@ -9,29 +9,27 @@
 
 - name: Get patch {{ patchnumber }} details
   uri:
-    url: "{{ dev_patcher_baseurl }}/?q=is:open+{{ patchnumber }}&o=ALL_REVISIONS"
+    url: "{{ dev_patcher_baseurl }}/?q={{ patchnumber }}&o=ALL_REVISIONS"
     return_content: yes
     body_format: json
   register: thispatch
 
 # Gerrit starts its http response with ")]}'". Strip that to have a valid json.
 - name: Parse patch details
   set_fact:
-    jsoncontent: "{{ thispatch.content.replace(pattern_to_replace,'') | from_json }}"
+    jsoncontent: "{{ thispatch.content.replace(pattern_to_replace,'') | from_json | first }}"
   vars:
     pattern_to_replace: ")]}'"
 
-- name: Apply {{ patchnumber }} - {{ jsoncontent[0]['subject'] }}
+- name: Apply {{ patchnumber }} - {{ jsoncontent['subject'] }}
   shell: |
     set -o errexit
-    git fetch {{ jsoncontent[0]['revisions'][revision]['fetch']['anonymous http']['url'] }} {{ jsoncontent[0]['revisions'][revision]['fetch']['anonymous http']['ref'] }}
+    git fetch {{ jsoncontent['revisions'][revision]['fetch']['anonymous http']['url'] }} {{ jsoncontent['revisions'][revision]['fetch']['anonymous http']['ref'] }}
     git cherry-pick FETCH_HEAD
   args:
     chdir: "{{ upstream_repos_clone_folder }}/{{ project }}"
   vars:
     # Use latest revision of no revision was specified
-    revision: "{{ (patchrevision != 'current') | ternary(patchrevision, jsoncontent[0]['current_revision']) }}"
-    project: "{{ jsoncontent[0]['project'] }}"
-  when:
-    # Do not apply patches if they are already merged
-    - "jsoncontent | length > 0"
+    revision: "{{ (patchrevision != 'current') | ternary(patchrevision, jsoncontent['current_revision']) }}"
+    project: "{{ jsoncontent['project'] }}"
+  changed_when: true
```

Comment:
Why we need to highlight that its **always** changed? Any specific reason ?

If repo content remains same as before this task (possible when there is no change in patch from what is present in local version), why we need to highlight that.

Reply:
- Without `changed_when`, lint complains
- I had `changed_when: false` before, but @evrardjp complained that it's wrong to say nothing has changed when we are doing changes
- And I probably agree that `changed_when: true` has more sense, because we are **always** patching

I'm not sure I see the situation when _repo content remains same as before this task_ . The reason of this whole change is, "if there's a patch that should be applied, apply it. If it fails something is wrong with the setup". So we always want to apply the patch that is in the list of patches